Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,583 @@
+ */
+gdjs.dialogueTree.getSelectedOption = function() {
+  if (this.dialogueData.select) {
+    return this.selectedOption;
+  }
+  return 0;
+};
+
+/**
+ * use this to check when the player has changed option selection.
+ * Use this to re-render your displayed dialogue options when needed.
+ */
+gdjs.dialogueTree.hasSelectedOptionChanged = function() {
+  if (this.selectedOptionUpdated) {
+    this.selectedOptionUpdated = false;
+    if (this.selectedOption === -1) this.selectedOption = 0;
+    return true;
+  }
+  return false;
+};
+
+/**
+ * use this to check the type of the Dialogue Line that is being displayed to the player at the moment.
+ * There are three types:
+ * - text - regular dialogue text is being parsed at the moment
+ * - options - the player has reached a branching choise moment where they must select one of multiple options
+ * - command - a <<command>> was called in the background, that can be used to trigger game events, but will not be displayed in the dialogue box.
+ * @param {string} type The type you want to check for ( one of the three above )
+ */
+gdjs.dialogueTree.isDialogueLineType = function(type) {
+  if (
+    this.commandCalls &&
+    this.commandCalls.some(function(call) {
+      return gdjs.dialogueTree.clipTextEnd > call.time;
+    })
+  ) {
+    return true;
+  }
+  return this.dialogueIsRunning ? this.dialogueDataType === type : false;
+};
+
+/**
+ * use this to check if a branch exists. It is also used internaly whenever you use the start from action.
+ * @param {string} branchName The Dialogue Branch name you want to check.
+ */
+gdjs.dialogueTree.hasDialogueBranch = function(branchName) {
+  return (
+    this.runner &&
+    this.runner.yarnNodes &&
+    Object.keys(this.runner.yarnNodes).some(function(node) {
+      return node === branchName;
+    })
+  );
+};
+
+/**
+ * Use this to start parsing dialogue from a specified Dialogue tree branch.
+ * Use this if you want to store multiple dialogues inside a single Dialogue tree data set.
+ * @param {string} startDialogueNode The Dialogue Branch name you want to start parsing from.
+ */
+gdjs.dialogueTree.startFrom = function(startDialogueNode) {
+  this.runner = gdjs.dialogueTree.runner;
+  if (!this.hasDialogueBranch(startDialogueNode)) return;
+  this.optionsCount = 0;
+  this.options = [];
+  this.dialogueBranchTitle = "";
+  this.dialogueBranchBody = "";
+  this.dialogueBranchTags = [];
+  this.tagParameters = [];
+  this.dialogue = this.runner.run(startDialogueNode);
+  this.dialogueData = null;
+  this.dialogueDataType = "";
+  this.dialogueText = "";
+  this.commandCalls = [];
+  this.commandParameters = [];
+  this.pauseScrolling = false;
+  this.dialogueData = this.dialogue.next().value; //Breaks here if json resource is used
+  this.dialogueIsRunning = true;
+  gdjs.dialogueTree.advanceDialogue();
+};
+
+/**
+ * Internal methods to check the type of a Dialogue Line
+ */
+gdjs.dialogueTree._lineTypeIsText = function() {
```

Comment:
nitpicking: `_isLineTypeText` (put is at the beginning) and same for the two other functions

Reply:
Btw, seems that these tools functions are a good idea, make things more readable!